FBSnapshotTestCase/FBSnapshotTestController.m

```diff
@@ -265,6 +265,8 @@ - (NSString *)_failedFilePathForSelector:(SEL)selector
   NSString *folderPath = NSTemporaryDirectory();
   if (getenv("IMAGE_DIFF_DIR")) {
     folderPath = @(getenv("IMAGE_DIFF_DIR"));
+  } else if ([_imageDiffDirectory length] > 0) {
+    folderPath = _imageDiffDirectory;
   }
```

Comment:
I can add this assert:

```
  NSAssert(getenv("IMAGE_DIFF_DIR") && _imageDiffDirectory.length, @"Image diff directory set in both environment variable and property");
```

But do you want to change the `if`/`else if` somehow as well?